internal/app/user-api/validation/user_validator.go

```diff
@@ -0,0 +1,78 @@
+package validation
+
+import (
+	"strings"
+
+	"github.com/afernandowex/ps-tag-onboarding-go/internal/app/model"
+	"github.com/afernandowex/ps-tag-onboarding-go/internal/app/user-api/constant"
+	"github.com/google/uuid"
+)
+
+type IUserValidationService interface {
+	ValidateUser(user *model.User) []string
+	ValidateUserID(ID *string) []string
+}
+
+type UserValidationService struct {
+}
+
+func (s *UserValidationService) ValidateUserID(ID *string) []string {
+	var errors []string
+	if ID == nil || !isUUID(*ID) {
+		errors = append(errors, constant.ErrorInvalidUserID)
+	}
+
+	return errors
+}
+
+func isUUID(str string) bool {
+	id, _ := uuid.Parse(str)
+
+	if id != uuid.Nil {
+		return true
+	} else {
+		return false
+	}
+}
+
+func (s *UserValidationService) ValidateUser(user *model.User) []string {
+	var errors []string
+	var err string = s.validateName(user)
+	if err != "" {
+		errors = append(errors, err)
```

Comment:
For future go knowledge, append on a slice allocates a whole new slice for just one element. If you have a use case with more than just 3 appends, you would want to pre allocate a slice and add to it.

Reply:
Yeah in this case, it is constant time which is why I resorted to slice.